packages/web/src/js/__tests__/Offcanvas.test.ts

```diff
@@ -0,0 +1,374 @@
+import Offcanvas from '../Offcanvas';
+import EventHandler from '../dom/EventHandler';
+import { clearFixture, getFixture } from '../../../tests/helpers/fixture';
+
+describe('Offcanvas', () => {
+  let fixtureEl: HTMLElement;
+
+  beforeAll(() => {
+    fixtureEl = getFixture();
+    // Dialog element do not work in Jest tests due to issue in jsdom
+    // `TypeError: this.element.showModal is not a function`
+    // @see: https://github.com/jsdom/jsdom/issues/3294
+    HTMLDialogElement.prototype.show = jest.fn();
+    HTMLDialogElement.prototype.showModal = jest.fn();
+    HTMLDialogElement.prototype.close = jest.fn();
+  });
+
+  afterEach(() => {
+    clearFixture();
+    jest.clearAllMocks();
+  });
+
+  describe('toggle', () => {
+    it('should call show method if show class is not present', () => {
+      fixtureEl.innerHTML = [
+        '<button id="btn" data-toggle="offcanvas" data-target="#offcanvas"></button>',
+        '<dialog class="offcanvas"></dialog>',
```

Comment:
how does it know it should open this dialog 😨 magic

Reply:
The `data-target` is useless here because the test works with the Offcanvas instance directly. And the target element is put manually inside.